components/telemetry-operator/internal/sync/sync.go

```diff
@@ -49,81 +54,91 @@ func NewLogPipelineSyncer(client client.Client,
 	return &lps
 }
 
-func (s *LogPipelineSyncer) SyncAll(ctx context.Context, logPipeline *telemetryv1alpha1.LogPipeline) (bool, error) {
+func (s *LogPipelineSyncer) SyncAll(ctx context.Context, logPipeline *telemetryv1alpha1.LogPipeline) (Result, error) {
+	var syncRes Result
 	log := logf.FromContext(ctx)
 
 	sectionsChanged, err := s.syncSectionsConfigMap(ctx, logPipeline)
 	if err != nil {
 		log.Error(err, "Failed to sync Sections ConfigMap")
-		return false, err
+		return syncRes, err
 	}
 
 	filesChanged, err := s.syncFilesConfigMap(ctx, logPipeline)
 	if err != nil {
 		log.Error(err, "Failed to sync mounted files")
-		return false, err
+		return syncRes, err
 	}
 	variablesChanged, err := s.syncVariables(ctx)
 	if err != nil {
 		log.Error(err, "Failed to sync variables")
-		return false, err
+		return syncRes, err
 	}
 
 	err = s.syncUnsupportedPluginsTotal(ctx)
 	if err != nil {
 		log.Error(err, "Failed to sync unsupported mode metrics")
-		return false, err
+		return syncRes, err
 	}
 
-	return sectionsChanged || filesChanged || variablesChanged, nil
+	if sectionsChanged.ConfigurationChanged || filesChanged.ConfigurationChanged || variablesChanged.ConfigurationChanged {
```

Comment:
Maybe replace the if statement by an an ||:
```
return Result{
  ConfigurationChanged: sectionsChanged.ConfigurationChanged || filesChanged.ConfigurationChanged || variablesChanged.ConfigurationChanged
  LogPipelineChanged: ... 
}
```

Not sure if the if / else logic of your solution is correct since both might change.

Reply:
done